pkg/table/tables/index.go

```diff
@@ -357,7 +356,10 @@ func (c *index) Create(sctx sessionctx.Context, txn kv.Transaction, indexedValue
 			}
 			continue
 		}
-		if c.idxInfo.Global && len(value) != 0 && !bytes.Equal(value, idxVal) {
+		// `insert ignore ... on duplicate update` / `update` may change a row's partition column.
+		// when a record row is modified from one partition to another partition,
+		// global index value(which contains partitionID) should be updated.
+		if c.idxInfo.Global && opt.CrossPartitionUpdate && len(value) != 0 && !bytes.Equal(value, idxVal) {
```

Comment:
Not get point here. Does it mean only `CrossPartitionUpdate = true` will update index?

Reply:
Yes, we should update global index row when the partition column is changed.